### PR TITLE
Add Gopkg dependency for kubernetes code-generator.

### PR DESCRIPTION
**Pull request by neuromage**
The code generator should not be run from HEAD, as it will generate code
that diverges from the pinned version of client-go, and also any
previously generated CRD controller clients.

This change pins both code generator and client-go to the specified
kubernetes release, and ensures the update-codegen.sh script uses the
code-generator specified in the vendor directory rather than HEAD. This
ensures the build is always reproducible.

Note: I made changes to Gopkg.toml and the update-codegen.sh script. The changes in vendor/ is the result of running `dep ensure`.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/371)

### CODE REVIEW

File: vendor/OWNERS [hunk 1]
```diff
@@ -1,10 +0,0 @@
-approvers:
```
Comment (IronPan) [inline vendor/OWNERS]: is this change intentional? if not is there any way to persist this file?

Comment (neuromage) [reply vendor/OWNERS]: Running `dep ensure` will delete that file, since all contents under vendor are assumed to be generated. I'm not sure how to keep that file in. Also I'm not sure if there's a benefit of having an OWNERS file in the vendor directory is, since changes under this directory should be a side effect of code, rather than a code change itself. What do you think?

**Comment by neuromage** [comment]
@vicaire and @IronPan can you take another look? I messed up some merges, which removed the lgtm, but finally managed to fix everything. PTAL thanks.
